### PR TITLE
Implementing frontend for editing client Demographics Info

### PR DESCRIPTION
**Pull request by joeldenning**

### CODE REVIEW

File: frontend/add-client/form-inputs/demographic-information-inputs.component.tsx [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+import CountrySelect from "../../util/country-select.component";
+import CurrencyInput from "../../util/currency-input.component";
+import { WeeklyEmployedHours } from "../../view-edit-client/view-client.component";
+
+export default function DemographicInformationInputs(
+  props: DemographicInformationInputsProps
+) {
+  const [civilStatus, setCivilStatus] = useState(CivilStatus.SINGLE);
+  const [householdIncome, setHouseholdIncome] = useState(
+    props.client.householdIncome
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply frontend/add-client/form-inputs/demographic-information-inputs.component.tsx]: Good question. I think leaving it empty is better -- the `<input>` is required by the form so they won't be able to proceed until they type in the real household income. This assures that they don't just gloss over the input and accept the default value without actually typing in the real one.

File: frontend/add-client/form-inputs/demographic-information-inputs.component.tsx [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+  const [householdSize, setHouseholdSize] = useState(
+    props.client.householdSize || 1
+  );
+  const [juvenileDependents, setJuvenileDependents] = useState(
+    props.client.juvenileDependents || 0
+  );
+  const [currentlyEmployed, setCurrentlyEmployed] = useState(
+    props.client.currentlyEmployed || "no"
+  );
+  const [weeklyEmployedHours, setWeeklyEmployedHours] = useState<
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply frontend/add-client/form-inputs/demographic-information-inputs.component.tsx]: No, the `<>` are [Typescript generics](https://www.typescriptlang.org/docs/handbook/generics.html)

File: frontend/add-client/form-inputs/demographic-information-inputs.component.tsx [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    WeeklyEmployedHours
+  >(props.client.weeklyEmployedHours || WeeklyEmployedHours["0-20"]);
+  const [employmentSector, setEmploymentSector] = useState(
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply frontend/add-client/form-inputs/demographic-information-inputs.component.tsx]: They do have defaults. See the next line where `getInitialEmploymentSector` is called

File: frontend/add-client/form-inputs/demographic-information-inputs.component.tsx [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+  const [payInterval, setPayInterval] = useState(
+    props.client.payInterval || PayInterval["every-two-weeks"]
+  );
+  const [countryOfOrigin, setCountryOfOrigin] = useState(
+    props.client.countryOfOrigin || "US"
+  );
+  const [dateOfUSArrival, setDateOfUSArrival] = useState(
+    props.client.dateOfUSArrival || ""
+  );
+  const [homeLanguage, setHomeLanguage] = useState(
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply frontend/add-client/form-inputs/demographic-information-inputs.component.tsx]: it does have a default. See next line where we call `getInitialHomeLanguage`

File: frontend/add-client/form-inputs/demographic-information-inputs.component.tsx [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+    currentlyEmployed,
+    employmentSector:
+      employmentSector === "other"
+        ? empSectorExplain
+        : employmentSector || null,
+    payInterval,
+    weeklyEmployedHours,
+    householdIncome,
+    householdSize,
+    isStudent,
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply frontend/add-client/form-inputs/demographic-information-inputs.component.tsx]: 👍 good catch. Will update.

File: frontend/add-client/form-inputs/demographic-information-inputs.component.tsx [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+    eligibleToVote,
+    registerToVote
+  };
+
+  return (
+    <form onSubmit={handleSubmit} autoComplete="off">
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply frontend/add-client/form-inputs/demographic-information-inputs.component.tsx]: This is the same order that it was before I moved the code. No rearranging happened in this PR. If we need to update the order, let's do it in a subsequent PR.

File: frontend/view-edit-client/view-edit-demographics-info.component.tsx [hunk 7]
```diff
@@ -146,6 +194,61 @@ export default function ViewEditDemographicsInfo(
...
+        homeLanguage: demographicsInfo.homeLanguage,
+        isStudent: demographicsInfo.isStudent,
+        englishProficiency: demographicsInfo.englishProficiency,
+        eligibleToVote: demographicsInfo.eligibleToVote,
+        registeredToVote: demographicsInfo.registeredToVote
+      }
+    });
+  }
+
+  function getDemographicsClient(): DemographicInformationClient {
```
Comment (MeghanBomberger) [inline frontend/view-edit-client/view-edit-demographics-info.component.tsx]: Does this also need to be changed to SingleClient?

Comment (MeghanBomberger) [inline frontend/view-edit-client/view-edit-demographics-info.component.tsx]: Does DemographicInformationClient  need to be changed to SingleClient here to match the change on 287

Comment (joeldenning) [reply frontend/view-edit-client/view-edit-demographics-info.component.tsx]: No. There are two slightly different definitions of a client -- the one in AddClient and the one in ViewEditClient. Both need to work with DemographicInformationClient, so DemographicInformationClient has its own definition of the properties it needs in order to work.

**Pull request merged**